### ISSUE TITLE
Missing User-Agent in Header

### ISSUE THREAD

**Issue by Tosox** [ClementJ18/moddb#52]
ModDB now requires the `User-Agent` field in the request headers. For example `moddb.login` will now fail because of [this](https://github.com/ClementJ18/moddb/blob/fe82ef91a355c4db597860be28ffc75629b4088f/moddb/utils.py#L227) request since `form` will result in `None`.

**Comment by Tosox** [comment]
![](https://github.com/user-attachments/assets/22ca3dbd-7910-4a9e-94ca-0f6e6293682f)

**Comment by ClementJ18** [comment]
Neither seem to work for me, I'll keep looking into this
```
>>> import requests
>>> requests.get("https://moddb.com")
<Response [403]>
>>> requests.get("https://moddb.com", headers={"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36"})
<Response [403]>
```

**Comment by Tosox** [comment]
Oh, that's annoying. I tried it on another device and got the same result as you. I tried to analyse the traffic with mitmproxy from the G.A.M.M.A. launcher (if you know what that is), which downloads a bunch of files from moddb, because it seems to have no problem with cloudflare. The requests are simple and have nothing special about them, but I haven't been able to replicate it so that it works even after mimicking the requests. However, I did manage to get a response using the cloudscraper library:
```
>>> import cloudscraper
>>> scraper = cloudscraper.create_scraper()
>>> r = scraper.get("https://www.moddb.com/members/login")
>>> r.status_code
200
>>> r.text[:500]
'<!DOCTYPE html>\r\n<html lang="en">\r\n<head>\r\n\t<title>Login - ModDB</title>\r\n\t<meta charset="utf-8">\r\n\t \r\n\t<meta name="keywords" content="login, log in, sign in, recover password, reset password, register, join, signup, help, session, cookie, navigation, development, community, contact us" />\r\n\t<meta name="description" content="Login to ModDB and enter a world of game development. Share your thoughts and ideas with the community." />\r\n\t<meta name="robots" content="max-image-preview:large" />\r\n\t<met'
```

**Comment by Tosox** [comment]
Okay, after trying things out and asking ChatGPT for some help I found another solution which worked for me:
```py
import requests
import ssl

# Define a custom adapter that allows using an SSL context
class MySSLAdapter(requests.adapters.HTTPAdapter):
    def init_poolmanager(self, *args, **kwargs):
        # Create a custom SSL context to limit TLS to 1.2 (and therefore disable TLS 1.3)
        ssl_context = ssl.create_default_context()
        ssl_context.maximum_version = ssl.TLSVersion.TLSv1_2 
        
        kwargs["ssl_context"] = ssl_context
        return super().init_poolmanager(*args, **kwargs)

url = "https://www.moddb.com/members/login"
headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Linux i686; rv:57.0) Gecko/20100101 Firefox/57.0',
}

session = requests.Session()
session.mount("https://", MySSLAdapter())

response = session.get(url, headers=headers)

print("Status Code:", response.status_code)
print("Response Text:", response.text[:500])
```

If the requests library uses TLS version 1.2 instead of 1.3 it works for me again.

**Comment by ClementJ18** [comment]
I've tried both of these and I'm getting very inconsistent results. Some page work a few times, some all the time and some not at all.

**Comment by ClementJ18** [comment]
Should hopefully be fixed with v0.12.0

**Comment by Tosox** [comment]
Thank you very much for your efforts :)
